### PR TITLE
fix: run callback when lint catch

### PR DESCRIPTION
**Pull request by fancyboynet**
This PR contains a:

- [x] **bugfix**

### CODE REVIEW

**Comment by fancyboynet** [comment]
#197

File: src/linter.js [hunk 1]
```diff
@@ -19,7 +19,9 @@ export default function linter(options, compiler, callback) {
         callback();
       }
     })
-    .catch(callback);
+    .catch(() => {
+      callback();
```
Comment (ricardogobbosouza) [inline src/linter.js]: We can put a condition here to fail if `failOnError` is enabled.
```js
.catch((e) => {
  if (options.failOnError) {
    callback(e);
  } else {
    callback();
  }
});
```

Comment (fancyboynet) [reply src/linter.js]: Yes,that is better

**Comment by fancyboynet** [comment]
@ricardogobbosouza How to do that? I have not change anything else.

**Comment by ricardogobbosouza** [comment]
@fancyboynet We put a condition and it must be tested. There are some tests that test this callback

**Comment by ricardogobbosouza** [comment]
If you prefer I can do it soon

**Comment by fancyboynet** [comment]
@ricardogobbosouza Yes,do it please.

**Comment by ricardogobbosouza** [comment]
@fancyboynet resolved on v1.1.1

**Pull request closed without merging**